File: OEAG/Perl/oeag.pl
#!/usr/bin/perl

use Digest::SHA qw(sha256);

my $auth   = $ARGV[0];

$auth=~s/\R//g;

if ( $auth eq "") {
	print "usage: oeag <authorization phrase>\n";
	exit 1;
}


my $sha64=unpack("H*", sha256($auth));

$sha64=~s/[^A-Za-z0-9]//g;

print $sha64;

1;

File: OEAG/Java/oeag-cli/src/oeag.java
import static org.apache.commons.codec.digest.MessageDigestAlgorithms.SHA_256;
import java.util.Base64;
import java.security.*;
import java.nio.charset.StandardCharsets;

public class oeag {

	public static void main(String[] args) throws NoSuchAlgorithmException {
		String plain;

		if (args.length > 0) {
			plain = args[0];

			System.out.println(plain);

			String sha256hex = org.apache.commons.codec.digest.DigestUtils.sha256Hex(plain);

			sha256hex = sha256hex.replaceAll("[^A-Za-z0-9]", "");

			System.out.println(sha256hex);

		} else {
			System.out.println("usage: oeag <authorization phrase>");
		}

	}

	public static String encode(final String clearText) throws NoSuchAlgorithmException {
		return new String(Base64.getEncoder()
				.encode(MessageDigest.getInstance("SHA-256").digest(clearText.getBytes(StandardCharsets.UTF_8))));
	}

}
